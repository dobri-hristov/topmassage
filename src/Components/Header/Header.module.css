.container {
    margin: 0;
    display: flex;
    flex-wrap: wrap;
    position: fixed;
    width: 100%;
    margin-top: -30px;
    padding: 0 20px;
    box-shadow: 0px 5px 10px 0px rgba(0, 0, 0, 0.75);
    z-index: 999;
}

@media screen and (max-width: 889px) {
    .container {
        display: none;
    }
}

.container p {
    margin: 5px;
}

.gray {
    background: #757594;
    color: white;
}

.black {
    background: black;
    color: white;
}