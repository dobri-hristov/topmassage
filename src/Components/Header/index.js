import React, { useState, useEffect } from 'react'
import styles from './Header.module.css'

const Header = () => {
    const [colors, setColors] = useState('gray')

    useEffect(() => {
        window.addEventListener('scroll', listenScrollEvent)
    })

    const listenScrollEvent = () => {
        window.scrollY < 160 ? setColors('gray') : setColors('black')
    }

    return (
        <div className={`${styles.container} ${styles[`${colors}`]}`}>
            <p><i className="fas fa-map-marker-alt"></i> град София, ж.к. Люлин, бул. "Царица Йоана" 49</p>
            <p><i className="fas fa-phone-alt"></i> +359 878 456 333</p>
            <p><i className="fas fa-envelope"></i> topmassage@gmail.com</p>
        </div>
    )
}

export default Header
