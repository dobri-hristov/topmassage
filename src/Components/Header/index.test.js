import React from 'react'
import TestRenderer from 'react-test-renderer'
import Header from './index.js'

describe('Header component', () => {
    test('should render correctly', () => {
        const component = TestRenderer.create(<Header />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})