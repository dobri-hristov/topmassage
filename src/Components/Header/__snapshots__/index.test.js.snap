// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div
  className="container gray"
>
  <p>
    <i
      className="fas fa-map-marker-alt"
    />
     град София, ж.к. Люлин, бул. "Царица Йоана" 49
  </p>
  <p>
    <i
      className="fas fa-phone-alt"
    />
     +359 878 456 333
  </p>
  <p>
    <i
      className="fas fa-envelope"
    />
     topmassage@gmail.com
  </p>
</div>
`;
